app/react/V2/Routes/Settings/Thesauri/EditThesaurus.tsx

```diff
@@ -0,0 +1,220 @@
+    }
+  }, [blocker, setShowNavigationModal]);
+
+  const addItem = () => {
+    setItemFormValues([]);
+    setShowThesauriValueFormSidepanel(true);
+  };
+  const addGroup = () => {
+    setGroupFormValues(emptyThesaurus());
+    setShowThesauriGroupFormSidepanel(true);
+  };
+  const editGroup = (row: Row<ThesaurusRow>) => {
+    setGroupFormValues(row.original);
+    setShowThesauriGroupFormSidepanel(true);
+  };
+  const editValue = (row: Row<ThesaurusRow>) => {
+    setItemFormValues([{ ...row.original, groupId: row.getParentRow()?.original.rowId }]);
+    setShowThesauriValueFormSidepanel(true);
+  };
+
+  const edit = (row: Row<ThesaurusRow>) => (row.original.subRows ? editGroup(row) : editValue(row));
+
+  const proceedDeletion = () => {
+    setThesaurusValues(prev => {
+      selectedThesaurusValue.forEach(deletedItem => {
+        removeItem(prev, deletedItem);
+      });
+      return [...prev];
+    });
+  };
+  const deleteSelected = () => {
+    if (!warnAboutUse || !selectedThesaurusValue[0].id) {
+      proceedDeletion();
+    } else {
+      setConfirmCallback({ callback: proceedDeletion });
+    }
+  };
+
+  return (
+    <div
+      className="tw-content"
+      style={{ width: '100%', overflowY: 'auto' }}
+      data-testid="settings-thesauri"
+    >
+      <SettingsContent className="flex flex-col h-full">
+        <SettingsContent.Header
+          path={new Map([['Thesauri', '/settings/thesauri']])}
+          title={watch('name')}
+        />
+        <SettingsContent.Body>
+          <ThesaurusForm
+            thesaurus={thesaurus}
+            edit={edit}
+            form={form}
+            thesaurusValues={thesaurusValues}
+            setThesaurusValues={setThesaurusValues}
+            setSelectedThesaurusValue={setSelectedThesaurusValue}
+          />
+        </SettingsContent.Body>
+        <SettingsContent.Footer className="bottom-0 bg-indigo-50">
+          {!isEmpty(selectedThesaurusValue) && (
+            <div className="flex items-center gap-2">
+              <Button
+                type="button"
+                onClick={deleteSelected}
+                color="error"
+                data-testid="thesauri-remove-button"
+              >
+                <Translate>Remove</Translate>
+              </Button>
+              <Translate>Selected</Translate> {selectedThesaurusValue.length}
+              <Translate>of</Translate> {thesaurusValues.length}
+            </div>
+          )}
+          {selectedThesaurusValue.length === 0 && (
+            <div className="flex justify-between w-full">
+              <div className="flex gap-2">
+                <Button onClick={addItem}>
+                  <Translate>Add item</Translate>
+                </Button>
+                <Button styling="light" onClick={addGroup}>
+                  <Translate>Add group</Translate>
+                </Button>
+                <Button styling="light" onClick={sortValues(thesaurusValues, setThesaurusValues)}>
```

Comment:
The table has it's own `sortingFn` prop that can be used to pass a custom sorting function. Could this not be used instead of a custom sorting function? Just rely on the table to run the sorting by title when this button is clicked? `app/react/V2/Routes/Settings/IX/IXSuggestions.tsx:252` does something like it.

Reply:
@Zasa-san I agree with you, but my attempt to use the sortingFn to update the sortingState from the table client turned into a more complex option, for sure I'm missing some part of the configuration, so for now I'll keeping the external sort function